### PR TITLE
TRUNK-5435: Improve Java 8 streams usage on Cohort.getActiveMembership(Patient)

### PR DESCRIPTION
**Pull request by Fungucide**
## Description of what I changed

Changed stream usage to filter->findFirst->get. Error handling might need to be changed as the function now returns a NoSuchElement error instead of the previous IndexOutOfBound. I'm still waiting for a response from a mentor on how I should deal with this issue.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5435

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you include the ticket id in your commit message as per? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by samuelmale** [comment]
Looks good to me. Can you also include a link to your GCI task instance?

**Comment by Fungucide** [comment]
https://codein.withgoogle.com/dashboard/task-instances/5258884047437824/

**Comment by suthagar23** [comment]
Hi @dkayiwa and @samuelmale  I have accepted this work since it was okay. 
@dkayiwa Can we go ahead to merge this?

File: api/src/main/java/org/openmrs/Cohort.java [hunk 1]
```diff
@@ -213,8 +213,7 @@ public boolean removeMembership(CohortMembership cohortMembership) {
 	 * @since 2.1.0
 	 */
 	public CohortMembership getActiveMembership(Patient patient) {
-		return getMemberships().stream().filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
-		        .collect(Collectors.toList()).get(0);
+		return getMemberships().stream().filter(CohortMembership::isActive).filter(m -> m.getPatientId().equals(patient.getPatientId())).findFirst().get();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Cohort.java]: Is there any reason why we are calling filter twice?

**Comment by suthagar23** [comment]
Hi @Fungucide , Please work for @dkayiwa  comment in this PR. you should complete this PR, and work to merge this before working on other tasks (BTW I have already approved your work).

File: api/src/main/java/org/openmrs/Cohort.java (hunk 1 above)
Comment (Fungucide) [reply api/src/main/java/org/openmrs/Cohort.java]: Hi, I was given some feedback on the Google Code In task page suggesting that I use function pointers for the filter predicate. Seeing how this is inefficient I will revert the code back to having just one filter.

**Pull request merged**